src/services/messengers/messenger_factory.py

```diff
@@ -1,12 +1,37 @@
-from typing import Dict, Tuple
+from typing import Dict, Tuple, Optional, Type, Callable
 from services.messengers.messenger import MessagingService
 from services.messengers.tg import TelegramMessenger
 from services.messengers.wa import WhatsappMessenger
 
-messenger_by_type = { 'tg' : TelegramMessenger(), 'wa' : WhatsappMessenger()} # type: Dict[str, MessagingService]
+messenger_by_type: Dict[str, Type[MessagingService]] = {'tg': TelegramMessenger, 'wa': WhatsappMessenger}
 
-def get_messenger_and_chat_id(messenger_chat_id:str) -> Tuple[MessagingService, str]:
+
+def make_messenger(messenger_chat_id: str) -> MessagingService:
     messenger_str, chat_id = messenger_chat_id.split(":")
-    messenger = messenger_by_type[messenger_str]
-    return messenger, chat_id
+    messenger = messenger_by_type[messenger_str](chat_id)
+    return messenger
+
+
+def _make_wa_messenger_from_event(event: Dict) -> Optional[MessagingService]:
```

Comment:
A bit ugly, probably would've been better to have:

chat_id = <>
make_messenger(type, chat_id)

For all functions, and then:

make_messenger_from_event
make_messenger_from_<> (for the other one)

but good enough for now :)